iceoryx_hoofs/vocabulary/include/iox/detail/expected.inl

```diff
@@ -14,17 +14,27 @@
 // limitations under the License.
 //
 // SPDX-License-Identifier: Apache-2.0
-#ifndef IOX_HOOFS_CXX_EXPECTED_INL
-#define IOX_HOOFS_CXX_EXPECTED_INL
+#ifndef IOX_HOOFS_VOCABULARY_EXPECTED_INL
+#define IOX_HOOFS_VOCABULARY_EXPECTED_INL
 
-#include "iceoryx_hoofs/cxx/expected.hpp"
+#include "iox/expected.hpp"
 
 namespace iox
 {
-namespace cxx
-{
 // AXIVION Next Construct AutosarC++19_03-A12.1.5 : This is a false positive since there is no fitting constructor
 // available for delegation
+namespace internal
+{
+template <typename... T>
+struct IsOptional : std::false_type
+{
+};
+template <typename T>
+struct IsOptional<iox::optional<T>> : std::true_type
+{
+};
+} // namespace internal
+
```

Comment:
I removed the above code.

Reply:
@FerdinandSpitzschnueffler Because in the meantime `IsOptional` was renamed to `is_optional`, see [here](https://github.com/eclipse-iceoryx/iceoryx/pull/1791#discussion_r1066275925).